services:
  app:
    build: 
      context: .
      dockerfile: Dockerfile
    ports:
      - "5000:5000"
    volumes:
      - ./app:/app
      - ./tests:/app/tests
    environment:
      - FLASK_APP=app.py
      - FLASK_DEBUG=1
      - DATABASE_URL=mysql+pymysql://mysql:mysql@db:3306/mysql
      - MYSQL_HOST=db
      - MYSQL_PORT=3306
      - MYSQL_DATABASE=mysql
      - MYSQL_USER=mysql
      - MYSQL_PASSWORD=mysql
      - PYTHONDONTWRITEBYTECODE=1
    depends_on:
      - db
    networks:
      flask-network:
        aliases:
          - flask-app

  db:
    image: mysql:8.0
    command: --default-authentication-plugin=mysql_native_password
    volumes:
      - mysql_data:/var/lib/mysql
      - ./db:/docker-entrypoint-initdb.d
    environment:
      - MYSQL_ROOT_PASSWORD=rootpassword
      - MYSQL_DATABASE=mysql
      - MYSQL_USER=mysql
      - MYSQL_PASSWORD=mysql
    ports:
      - "3306:3306"
    networks:
      - flask-network

  selenium:
    image: selenium/standalone-chrome:latest
    shm_size: 2gb
    environment:
      - SE_NODE_MAX_SESSIONS=1
      - SE_NODE_OVERRIDE_MAX_SESSIONS=true
    ports:
      - "4444:4444"
    networks:
      - flask-network
    profiles:
      - test

  ui-tests:
    build:
      context: .
      dockerfile: Dockerfile
    entrypoint: ["pytest"]
    command: ["tests/ui", "--maxfail=1", "--disable-warnings"]
    working_dir: /workspace
    volumes:
      - .:/workspace
    environment:
      - APP_URL=http://flask-app:5000
      - SELENIUM_REMOTE_URL=http://selenium:4444/wd/hub
    networks:
      - flask-network
    profiles:
      - test
    depends_on:
      - app
      - selenium

volumes:
  mysql_data:

networks:
  flask-network:
    driver: bridge
